src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -3424,103 +3452,107 @@ class StubGenerator: public StubCodeGenerator {
     Register rscratch3 = r10;
     Register rscratch4 = r11;
 
+    Register state_regs[2] = { r12, r13 };
+    RegSet saved_regs = RegSet::range(r16, r22) - r18_tls;
+    Cached64Bytes reg_cache(_masm, buf, RegSet::of(r14, r15) + saved_regs);
+
```

Comment:
Perhaps add a note here to the effect that the rest of this patch requires there to be **exactly** 8 registers in this set. Maybe assert that here?

Reply:
This requirement has been asserted in the constructor.

Reply:
Sure, but a note here would have made it easier to understand.